File: ruby-pr/53/ruby/code-quality-readability-maintainability/value_calculator.rb
class ValueCalculator
  def calculate_total(price, quantity, customer_type, has_discount)
    # {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
    total = price * quantity
    total = total * 0.9 if customer_type == 'VIP'
    total = total * 0.95 if customer_type == 'Regular'
    total = total * 0.85 if has_discount
    total = total * 0.98 if total > 1000
    total = total * 0.97 if quantity > 10
    total
    # {/fact}
  end
end

File: php-pr/51/php/code-quality-readability-maintainability/ValueCalculator.php
<?php

class ValueCalculator
{
    public function calculateTotal($price, $quantity, $customerType, $hasDiscount)
    {
        // {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
        $total = $price * $quantity;
        if ($customerType == 'VIP') $total = $total * 0.9;
        else if ($customerType == 'Regular') $total = $total * 0.95;
        if ($hasDiscount) $total = $total * 0.85;
        if ($total > 1000) $total = $total * 0.98;
        if ($quantity > 10) $total = $total * 0.97;
        return $total;
        // {/fact}
    }
}